dask/blockwise.py

```diff
@@ -243,6 +251,14 @@ def _out_numblocks(self):
 
         return out_d
 
+    def get_dependencies(self, all_hlg_keys):
+        _ = self._dict  # trigger materialization
+        return self._cached_dict["basic_layer"].get_dependencies(all_hlg_keys)
+
+    def cull(self, keys, all_hlg_keys):
+        _ = self._dict  # trigger materialization
+        return self._cached_dict["basic_layer"].cull(keys, all_hlg_keys)
```

Comment:
I think that in the future we may just want to track the desired output keys, 

```python
layer = Blockwise(..., actually_desired_keys=[("x", 0, 1), ("x", 20, 12)])
```

Then when we call things like `layer.get_dependencies` we check in with that set of actually desired keys (hopefully we find a better name) and generate the dependencies on the fly.

Reply:
This can be safely postponed for future work though.

Reply:
Yes, this is also something like that I had in mind. But it is not trivial to do without ending up generating more-or-less the full graph.